utils/torch_solver.py

```diff
@@ -1,5 +1,6 @@
 import os
 import sys
+from tqdm import tqdm
```

Comment:
Maybe we can move this in the safe import context?

Reply:
yes, we should :)